src/test/java/com/commercetools/payment/handler/CommercetoolsCreatePaymentWithExperienceProfileIdIT.java

```diff
@@ -0,0 +1,162 @@
+import static java.lang.String.format;
+import static java.util.Optional.ofNullable;
+import static javax.swing.Action.DEFAULT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.http.MediaType.APPLICATION_JSON_UTF8;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Test payment creation with web experience profile id.
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = Application.class)
+@AutoConfigureMockMvc
+// 1) completely wipe-out CTP project Payment, Cart, Order endpoints before the test cases
+// 1) ensure noAddressOverrideWebProfile profile exists
+@Import({OrdersCartsPaymentsCleanupConfiguration.class, WebProfileConfiguration.class})
+public class CommercetoolsCreatePaymentWithExperienceProfileIdIT extends PaymentIntegrationTest {
+
+    @Autowired
+    private MockMvcAsync mockMvcAsync;
+
+    @Autowired
+    private TenantConfigFactory tenantConfigFactory;
+
+    @Autowired
+    private CtpFacadeFactory ctpFacadeFactory;
+
+    @Autowired
+    private SphereClientFactory sphereClientFactory;
+
+    /**
+     * From {@link WebProfileConfiguration}
+     */
+    @Autowired
+    private WebProfile noAddressOverrideWebProfile;
+
+    private TenantConfig tenantConfig;
+    private SphereClient sphereClient;
+    private CtpFacade ctpFacade;
+
+    @Before
+    public void setUp() throws Exception {
+        tenantConfig = tenantConfigFactory.getTenantConfig(MAIN_TEST_TENANT_NAME)
+                .orElseThrow(IllegalStateException::new);
+        ctpFacade = ctpFacadeFactory.getCtpFacade(tenantConfig);
+        sphereClient = sphereClientFactory.createSphereClient(tenantConfig);
+    }
+
+    @Test
+    public void paymentWithExperienceProfileIdCreated() throws Exception {
+        final String ctpPaymentId = createCartAndPayment(sphereClient);
+        MvcResult mvcResult = mockMvcAsync.performAsync(post(format("/%s/commercetools/create/payments/%s", MAIN_TEST_TENANT_NAME, ctpPaymentId)))
+                .andDo(print())
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(APPLICATION_JSON_UTF8))
+                .andReturn();
+
+        // validate response json: approvalUrl
+        String returnedApprovalUrl = verifyApprovalUrl(mvcResult);
+
+        // verify CTP payment values: approval url + interfaceId + interface interaction
+        Payment updatedPayment = executeBlocking(ctpFacade.getPaymentService().getById(ctpPaymentId)).orElse(null);
+        assertThat(updatedPayment).isNotNull();
+
+        assertThat(getCustomFieldStringOrEmpty(updatedPayment, APPROVAL_URL)).isEqualTo(returnedApprovalUrl);
```

Comment:
Are these asserts necessary? This test checks if xp profile id is created, so I think we should focus only on that. Other tests can check if payment is created correctly. It's like in a method - one method do only one thing.

Reply:
since it is an integration test - i want to ensure completely the created payment state